Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/ui/details/RecoveryCertificateDetailsFragment.kt

```diff
@@ -0,0 +1,115 @@
+        appBarLayout.onOffsetChange { titleAlpha, subtitleAlpha ->
+            title.alpha = titleAlpha
+            subtitle.alpha = subtitleAlpha
+        }
+
+        bindToolbar()
+        setToolbarOverlay()
+
+        viewModel.qrCode.observe(viewLifecycleOwner) { onQrCodeReady(it) }
+        viewModel.errors.observe(viewLifecycleOwner) { onError(it) }
+        viewModel.events.observe(viewLifecycleOwner) { onNavEvent(it) }
+        viewModel.recoveryCertificate.observe(viewLifecycleOwner) { it?.let { onCertificateReady(it) } }
+    }
+
+    private fun FragmentRecoveryCertificateDetailsBinding.onCertificateReady(
+        recoveryCertificate: RecoveryCertificate
+    ) {
+        /* TODO */
+    }
+
+    private fun FragmentRecoveryCertificateDetailsBinding.onQrCodeReady(bitmap: Bitmap?) {
+        qrCodeCard.apply {
+            image.setImageBitmap(bitmap)
+            progressBar.hide()
+            bitmap?.let { image.setOnClickListener { viewModel.openFullScreen() } }
+        }
+    }
+
+    private fun FragmentRecoveryCertificateDetailsBinding.onError(error: Throwable) {
+        qrCodeCard.progressBar.hide()
+        error.toErrorDialogBuilder(requireContext()).show()
+    }
+
+    private fun FragmentRecoveryCertificateDetailsBinding.onNavEvent(event: RecoveryCertificateDetailsNavigation) {
+        when (event) {
+            RecoveryCertificateDetailsNavigation.Back -> popBackStack()
+            is RecoveryCertificateDetailsNavigation.FullQrCode -> findNavController().navigate(
+                R.id.action_global_qrCodeFullScreenFragment,
+                QrCodeFullScreenFragmentArgs(event.qrCodeText).toBundle(),
+                null,
+                FragmentNavigatorExtras(qrCodeCard.image to qrCodeCard.image.transitionName)
+            )
+        }
+    }
+
+    private fun FragmentRecoveryCertificateDetailsBinding.bindToolbar() = toolbar.apply {
+        setNavigationOnClickListener { popBackStack() }
+        setOnMenuItemClickListener {
+            when (it.itemId) {
+                R.id.menu_covid_certificate_delete -> {
+                    DialogHelper.showDialog(deleteTestConfirmationDialog)
+                    true
+                }
+                else -> onOptionsItemSelected(it)
+            }
+        }
+    }
+
+    private fun setToolbarOverlay() {
+        val width = requireContext().resources.displayMetrics.widthPixels
+        val params: CoordinatorLayout.LayoutParams = binding.scrollView.layoutParams as (CoordinatorLayout.LayoutParams)
+
+        val textParams = binding.subtitle.layoutParams as (LinearLayout.LayoutParams)
+        textParams.bottomMargin = (width / 3) + 170
+        binding.subtitle.requestLayout()
+
+        val behavior: AppBarLayout.ScrollingViewBehavior = params.behavior as (AppBarLayout.ScrollingViewBehavior)
+        behavior.overlayTop = (width / 3) + 170
+    }
+
+    private val deleteTestConfirmationDialog by lazy {
+        DialogHelper.DialogInstance(
```

Comment:
I think latest trend is to use `MaterialAlertDialogBuilder` directly , can you please use it ?